pkg/kubefed2/federate/federate.go

```diff
@@ -188,24 +183,28 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 	}, nil
 }
 
-func lookupTypeDetails(config *rest.Config, qualifiedTypeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
-	client, err := genericclient.New(config)
+func lookupTypeDetails(hostConfig *rest.Config, typeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
+	apiResource, err := enable.LookupAPIResource(hostConfig, typeName.Name, "")
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to get federation client")
+		return nil, err
 	}
+	typeName.Name = typeconfig.GroupQualifiedName(*apiResource)
 
-	typeConfig := &fedv1a1.FederatedTypeConfig{}
-	err = client.Get(context.TODO(), typeConfig, qualifiedTypeName.Namespace, qualifiedTypeName.Name)
+	client, err := genericclient.New(hostConfig)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Error retrieving FederatedTypeConfig %q", qualifiedTypeName)
+		return nil, errors.Wrap(err, "Failed to get federation client")
 	}
 
-	_, err = enable.LookupAPIResource(config, typeConfig.Name, typeConfig.APIVersion)
+	typeConfig := &fedv1a1.FederatedTypeConfig{}
+	err = client.Get(context.TODO(), typeConfig, typeName.Namespace, typeName.Name)
+	if apierrors.IsNotFound(err) {
+		return nil, errors.Wrapf(err, "FederatedTypeConfig %q not found. Try 'kubefed2 enable type %s' before federating the resource", typeName, typeName.Name)
```

Comment:
(No action required) Wrapping the error makes the first part of the error message redundant.  I'll make sure this gets fixed by the end of the series but its non-functional for now.